src/tui.rs

```diff
@@ -734,37 +1412,143 @@ fn ui(f: &mut Frame, app: &App) {
+            .style(Style::default().fg(Color::Gray))
+            .alignment(Alignment::Center);
+        f.render_widget(placeholder, area);
+        return;
+    }
 
-    // Bar chart for visual allocation
-    let data: Vec<(&str, u64)> = allocation_vec
-        .iter()
-        .map(|(name, percentage)| (name.as_str(), **percentage as u64))
-        .collect();
+    // Find min/max values for scaling
+    let min_value = historic_data.iter().map(|(_, v)| *v).fold(f64::INFINITY, f64::min);
+    let max_value = historic_data.iter().map(|(_, v)| *v).fold(f64::NEG_INFINITY, f64::max);
+    let min_week = historic_data.iter().map(|(d, _)| *d).fold(f64::INFINITY, f64::min);
+    let max_week = historic_data.iter().map(|(d, _)| *d).fold(f64::NEG_INFINITY, f64::max);
 
-    let barchart = BarChart::default()
+    // Add some padding to the value range
+    let value_padding = (max_value - min_value) * 0.1;
+    let y_min = (min_value - value_padding).max(0.0);
+    let y_max = max_value + value_padding;
+
+    let datasets = vec![Dataset::default()
+        .marker(ratatui::symbols::Marker::Braille)
+        .style(Style::default().fg(Color::Cyan))
+        .graph_type(GraphType::Line)
+        .data(&historic_data)];
+
+    let chart = Chart::new(datasets)
         .block(
             Block::default()
                 .borders(Borders::ALL)
-                .title("Asset Allocation"),
+                .title("Portfolio Growth")
         )
-        .data(&data)
-        .bar_width(9)
-        .bar_style(Style::default().fg(Color::Yellow))
-        .value_style(Style::default().fg(Color::Black).bg(Color::Yellow));
+        .x_axis(
+            Axis::default()
+                .title("Weeks Since First Purchase")
+                .style(Style::default().fg(Color::Gray))
+                .bounds([min_week, max_week])
+                .labels(vec![
+                    Line::from("0"),
+                    Line::from(format!("{:.0}", max_week * 0.25)),
+                    Line::from(format!("{:.0}", max_week * 0.5)),
+                    Line::from(format!("{:.0}", max_week * 0.75)),
+                    Line::from(format!("{:.0}", max_week)),
+                ])
+        )
+        .y_axis(
+            Axis::default()
+                .title(app.currency.as_str())
+                .style(Style::default().fg(Color::Gray))
+                .bounds([y_min, y_max])
+                .labels(vec![
+                    Line::from(format!("{:.0}", y_min)),
+                    Line::from(format!("{:.0}", (y_min + y_max) * 0.25)),
+                    Line::from(format!("{:.0}", (y_min + y_max) * 0.5)),
+                    Line::from(format!("{:.0}", (y_min + y_max) * 0.75)),
+                    Line::from(format!("{:.0}", y_max)),
+                ])
+        );
 
-    f.render_widget(barchart, area);
+    f.render_widget(chart, area);
+}
+
+fn render_pie_chart(f: &mut Frame, area: Rect, portfolio: &Portfolio) {
```

Comment:
This function is wrongly named.

Also, the "Asset Breakdown" and "Detailed Allocation" contain the wrong data.

The "Detailed Allocation" should list all assets, while the Asset Breakdown should group them by asset.